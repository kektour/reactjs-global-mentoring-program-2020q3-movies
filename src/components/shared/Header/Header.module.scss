@import 'src/shared.scss';

.root {
}

.container {
  @extend %container;
}

.header {
  display: flex;
  justify-content: space-between;
}
