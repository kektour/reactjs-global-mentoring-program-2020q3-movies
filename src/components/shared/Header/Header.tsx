import React from 'react';
import classnames from 'classnames';
import { Logo } from '../Logo';
import styles from './Header.module.scss';

type Props = {
  classes?: {
    root?: string;
  };
};

export const Header: React.FC<Props> = (props) => {
  const { classes = {}, children } = props;
  return (
    <div className={classnames(styles.root, classes.root)}>
      <div className={styles.container}>
        <div className={styles.header}>
          <Logo />
          {children}
        </div>
      </div>
    </div>
  );
};
