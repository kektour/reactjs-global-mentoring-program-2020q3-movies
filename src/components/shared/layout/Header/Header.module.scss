@import 'src/shared.scss';

.root {
  display: flex;
  justify-content: space-between;
}

.button {
  @extend %button;
  height: 50px;
  color: #f65261;
  background: rgba(85, 85, 85, 0.6);
}
