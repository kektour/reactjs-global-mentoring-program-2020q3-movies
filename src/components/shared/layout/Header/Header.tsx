import React from 'react';
import { Logo } from '../Logo';
import styles from './Header.module.scss';

type Props = {
  onButtonClick?: () => void;
};

export const Header: React.FC<Props> = (props) => {
  const { onButtonClick: handleButtonClick = () => null } = props;
  return (
    <div className={styles.root}>
      <Logo />
      <button className={styles.button} onClick={handleButtonClick}>
        + Add Movie
      </button>
    </div>
  );
};
