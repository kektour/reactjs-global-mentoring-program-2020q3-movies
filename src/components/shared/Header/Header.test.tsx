import React from 'react';
import { render } from '@testing-library/react';
import { Header } from './Header';

describe('Header', () => {
  it('Should render component correctly', () => {
    const component = render(<Header />);
    expect(component.asFragment()).toMatchSnapshot();
  });
});
