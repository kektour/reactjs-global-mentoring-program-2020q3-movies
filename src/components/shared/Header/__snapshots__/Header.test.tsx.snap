// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should render component correctly 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="container"
    >
      <div
        class="header"
      >
        <p
          class="root"
        >
          <span>
            netflix
          </span>
          roulette
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
